### PR TITLE
git-latexdiff: migrate from core

### PR DESCRIPTION
**Pull request by L2G**
See also https://github.com/Homebrew/homebrew/pull/39463.

### CODE REVIEW

**Comment by xu-cheng** [comment]
* Need `depends_on :tex`
* Need to fix all audit --strict errors
* Can you write a better test as the TeX is available in this repo's ci environment? Thanks

**Comment by L2G** [comment]
Yup. These changes were coming next; I just wasn't sure if it was OK to conflate them with the migration. (Sounds like it is.)

**Comment by L2G** [comment]
git-latexdiff is a wrapper for latexdiff, but latexdiff is apparently not on the test system, so I'm blocked.

**Comment by xu-cheng** [comment]
Looks like `latexdiff` is available in MacTeX but not BasicTeX.

You can always create a `latexdiff` formula.

**Comment by L2G** [comment]
But anyone using Homebrew on a Mac is going to be advised by Homebrew to install MacTeX, and as you said, latexdiff is available in MacTex.

**Comment by xu-cheng** [comment]
Generally we would recommend users to install MacTeX. (In fact there're formulae in homebrew/tex which are available in MacTeX e.g. latexmk).

I would like think homebrew/tex is a supplement of MacTeX/BasicTeX. That is we offer formulae with latest version OR offer formulae that are not available in BasicTeX (So it can benefit users who are unwilling to install MacTeX e.g. small disk quota)

**Comment by L2G** [comment]
I see. In that case, would it be okay to make it `depends_on "latexdiff" => :recommended`?

**Comment by xu-cheng** [comment]
> In that case, would it be okay to make it depends_on "latexdiff" => :recommended?

It's OK.

**Comment by L2G** [comment]
Actually, would it be more correct to use `depends_on "homebrew/tex/latexdiff" => :recommended`?

**Comment by xu-cheng** [comment]
No, we only use full qualify name for cross tap dependency. And these two formulae are in the same tap.

File: Formula/git-latexdiff.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class GitLatexdiff < Formula
+  homepage "https://gitorious.org/git-latexdiff"
+  url "https://gitorious.org/git-latexdiff/git-latexdiff.git",
+    :tag => "v1.0.0",
+    :revision => "c3f74f5559867111f118600e6d2bf778a3f13754"
+
+  option "without-latexdiff", "Don't install latexdiff (in case your TeX "\
+                              "distro already has it)"
```
Comment (xu-cheng) [inline Formula/git-latexdiff.rb]: This is unnecessary. (I will remove it when merging)

**Comment by xu-cheng** [comment]
Thanks for the pull request! 🎉 Homebrew depends on contributions from community members like you and we're grateful for your support.

File: Formula/git-latexdiff.rb (hunk 1 above)
Comment (L2G) [inline Formula/git-latexdiff.rb]: The line was there mainly to reword the default, generic description of `--without-latexdiff`.

**Pull request closed without merging**